File command documentation

mkothbauer opened the pull request:
# Advanced HW 9 submission

Closes #83

Name: Melinda Kothbauer
Uniqname: mkothbau

the-alex commented on _commands/tools/file.md:
```diff
@@ -0,0 +1,42 @@
+
+#### Break it down
+
+* `-i' changes the way that the command interprets file types, giving more information about the file
```
I think you meant `i` instead of `-i'

the-alex commented on _commands/tools/file.md:
```diff
@@ -0,0 +1,42 @@
+
+### Useful Options/Examples
+
+### `file -F, --separator _separator_`
```
I think I understand that the `,` is trying to separate the command flags from the command arguments, but it's not obvious that this is the case. I would rather them just be removed- I think it's clear enough without them.

Additionally, the redundancy of the term "separator" twice (`--seperator`, `_separator_`), but only referring to one instance of it (`_separator_`) is also a little confusing. I just want to make sure the first instance (`--separator`) isn't a type, because it doesn't look like you include the word after the two dashes in the code example below.

mkothbauer commented:
I updated my file based on the comments you made. Let me know if there is anything else!

ppannuto commented:
Sorry to chime in late and a little contradictory, @the-alex other articles say things like "the `-i` flag"..., which I think is clearer because then people can pattern match the string `-i` exactly to the command instead of mentally stripping the `-` (Don't strongly feel this way, but it is the convention we started); would you mind flipping that line back? (Maybe add the wording "`-i` flag")

Also this needs a `<!--more-->` break; see the other merged commands for an example.

mkothbauer commented:
Okay, I changed it to '-F flag' and '-i flag' - I think that is the clearest. Additionally, I added some more clarification on the _separator_ argument. Also added the <!--more--> as requested.

ppannuto commented:
The last thing I'd love to see is a little bit between the "more" break and before the useful commands that explains how file works. The whole mechanism is a little complex, but the simple version that's useful here is to explain that many files have a "magic" header, the first few bytes of the file that explain what kind of file it is, for example, PNG's happens to be human-readable:

```bash
$ hexdump -C window.png | head -1
00000000  89 50 4e 47 0d 0a 1a 0a  00 00 00 0d 49 48 44 52  |.PNG........IHDR|
```

The bytes `#!` are another form of magic header, which is how the operating system knows to take files that start with `#!/bin/bash` or `#!/usr/bin/env python3` and interpret them correctly when you run `./my_script`.

Also point to `man 5 magic` for more info on this.

If there is no magic header, (as is the case in your examples here with `file.md`), then it will scan the rest of the document to make an educated guess (e.g. the file `file.md` doesn't have any bytes in the range 0x00-0x20, since those don't appear in ascii text, making text a good guess).

The point of this explanation is to explain that file can be imperfect, and to give some intuition for how it works.

ppannuto commented:
Final nit from me is that you can't use `<i>` in code blocks, you'd have to do something like this:

```
-### `file -F <i>separator</i>`
+### `file -F `_`separator`_
```

to make it work. Otherwise +1 from me, I'll leave it for any final comments from @the-alex and let him merge.

mkothbauer commented:
Yes! Thanks for the catch on the <i> markings. Is there any way to combine italics and code in markdown? Googling told me that the <i> flag would work, but clearly it did not.

ppannuto commented:
Yeah, it's a bit of a hack, but you essentially put the modifiers outside of
the code block and it'll do the right thing, I believe that either of

  

```

`in code block `_`italicized words`_`the rest`

`in code block `&lt;i&gt;`italicized words`&lt;/i&gt;` the rest`

```

  

will work. With `jekyll serve` running, the site will rebuild every time you
save the file. Just play with it in a text editor / refreshing the browser
until it looks like what you want :)

> On Mar 21 2016, at 11:55 am, Melinda Kothbauer
&lt;notifications@github.com&gt; wrote:  

>

> Yes! Thanks for the catch on the _ markings. Is there any way to combine
italics and code in markdown? Googling told me that the _ flag would work, but
clearly it did not. __

>

> —  
You are receiving this because you commented.  
Reply to this email directly or [view it on GitHub](https://github.com/c4cs/c4
cs.github.io/pull/90#issuecomment-199353295)![](https://github.com/notificatio
ns/beacon/AAUt3ptn2M4hgWxJxIIcWk9AKb7BBuCWks5pvr-JgaJpZM4H0zKm.gif)

mkothbauer commented:
Cool, thank you!

the-alex commented:
Oh, @ppannuto , I wasn't saying to leave out the hyphen in `-i`, I was pointing out that there wasn't a second backtick, so the inline code wasn't rendering. It rendered, literally, as  `-i' instead of `-i`, which is what I assumed she meant. That got fixed anyways, so it doesn't matter!

the-alex commented on _commands/tools/file.md:
```diff
@@ -0,0 +1,64 @@
+
+### `file -i`
+### `file --mime`
+
```
![screen shot 2016-03-22 at 10 26 39 pm](https://cloud.githubusercontent.com/assets/7095187/13973856/52a4889e-f07d-11e5-8c03-87f13916d12c.png)

Looks a 'lil funny there.

ppannuto commented:
Looks good, thanks!
